lib/markdown_videos/services/service_base.rb

```diff
@@ -0,0 +1,124 @@
+module MarkdownVideos
+  module Services
+
+    class ServiceBase
+
+      attr_accessor :id,
+                    :alt_text,
+                    :markdown_url,
+                    :width,
+                    :height,
+                    :wrapper,
+                    :classname
+
+      def self.support(markdown_url)
+        markdown_url =~ regexp
+      end
+
+      def self.regexp
+        raise "#{self.class} class should respond to `regexp`"
+      end
+
+
+      def initialize(alt_text, markdown_url, options)
+        @alt_text = alt_text
```

Comment:
Because `alt_text` is mandatory by the syntax itself, even if empty, `[]` needs to be there.